### PR TITLE
public API: Add new endpoints

### PR DESCRIPTION
**Pull request by tahini**
Add endpoints to get the agencies, services and lines

Add an endpoint to get the supported versions

Add the x-added-in-version extension to the yaml documentation

### CODE REVIEW

File: packages/transition-backend/src/api/__tests__/public.routes.test.ts [hunk 1]
```diff
@@ -360,82 +423,174 @@ describe('Testing API endpoints', () => {
...
-            (calculateAccessibilityMap as jest.Mock).mockResolvedValue(result);
+            mockedCalculateAccessibilityMap.mockResolvedValue(calculationResult);
             (TransitAccessibilityMapRouting as any).mockImplementation(() => mockTransitAccessibilityMapRouting);
-            (AccessibilityMapAPIResponse as any).mockImplementation((input) => ({
-                getResponse: () => input.resultParams
-            }));
 
             const response = await request(app)
-                .post('/api/v1/accessibility?withGeojson=false')
+                .post(`/api/v1/accessibility?withGeojson=false`)
```
Comment (greenscientist) [inline packages/transition-backend/src/api/__tests__/public.routes.test.ts:573]: Might not need to change to backticks

File: packages/transition-backend/src/api/__tests__/public.routes.test.ts [hunk 2]
```diff
@@ -360,82 +423,174 @@ describe('Testing API endpoints', () => {
...
+            };
+
+            const expectedOutput = {
+                result: {
+                    nodes: resultByNode.nodes,
+                    polygons: {
+                        type: 'FeatureCollection',
+                        features: [{
+                            type: 'Feature',
+                            geometry: calculationResult.polygons.features[0].geometry,
```
Comment (greenscientist) [inline packages/transition-backend/src/api/__tests__/public.routes.test.ts:614]: Probably fine, but if the feature change somehow, we won't detect the change

File: docs/APIv1/API.yml [hunk 3]
```diff
@@ -72,6 +72,18 @@ paths:
               schema:
                 $ref: 'scenarios.yml#/scenarios'
 
+  /api/v1/agencies:
+    get:
+      summary: Get all agencies
+      description: Get all agencies which are currently available in Transition
```
Comment (greenscientist) [inline docs/APIv1/API.yml:95]: Do you want to add (and bump the X-version tag)

Comment (greenscientist) [reply docs/APIv1/API.yml:95]: ok, was added in next commit

File: packages/transition-backend/src/api/__tests__/public.routes.test.ts [hunk 4]
```diff
@@ -825,6 +825,14 @@ describe('Testing API endpoints', () => {
         });
     });
 
+    test('/api/v1/version endpoint', async () => {
+
+        const response = await request(app).get('/api/versions');
+
+        expect(response.status).toStrictEqual(200);
+        expect(response.body).toStrictEqual(['1.1']);
```
Comment (greenscientist) [inline packages/transition-backend/src/api/__tests__/public.routes.test.ts:877]: Do we want to mock the version or we want to change the test with each version bump?

**Review by greenscientist — COMMENTED** [review]
Nothing major so far

File: packages/transition-backend/src/api/__tests__/public.routes.test.ts (hunk 4 above)
Comment (tahini) [reply packages/transition-backend/src/api/__tests__/public.routes.test.ts:877]: we want to be aware of a version change and make sure it is correctly returned, so it's hard-coded. I'll add a comment.

**Review by greenscientist — APPROVED** [review]

**Pull request merged**